pkg/controller/instance/instance_controller.go

```diff
@@ -328,6 +331,20 @@ func createPlan(mgr manager.Manager, planName string, instance *kudov1alpha1.Ins
 		Namespace: instance.Namespace,
 	}
 
+	// first lets check if there is an active plan
+	if instance.Status.ActivePlan.Namespace != "" {
+		// So you're saying there's a chance...
+		pe := &kudov1alpha1.PlanExecution{}
+		err := mgr.GetClient().Get(ctx, client.ObjectKey{
```

Comment:
do we really need to do the GET request here? We cannot trust that when activeplan is set that planexecution exists? I think that must be true, if activeplan is set, then corresponding planexecution exists

Reply:
It is needed yes...

Reply:
The case this solves is when the manager restarts.   when it does up again, the `createPlan` is called for existing instances.   I struggle if that was the right thing and decided that it was ok.   createPlan perhaps could be renamed to establishPlan :)      When createPlan is originally invoked, it doesn't have an ActivePlan... so this isn't called.

Reply:
I still don't understand why is the get needed, can you please explain when is ActivePlan on instance set but the planexecution does not exist?

Reply:
ActivePlan on InstanceStatus is an object ref not a ptr: https://github.com/kudobuilder/kudo/blob/master/pkg/apis/kudo/v1alpha1/instance_types.go#L35
The construction of InstanceStatus will default ActivePlan. which will default types in it (mostly strings which will be "").   This results in the situation that if there is every an InstanceStatus... there is always an instance of ActivePlan (even though one wasn't explicitly initialized).   It should be expected that if there is an ActivePlan it has a name.   Hence the `instance.Status.ActivePlan.Name != ""` meaning no active plan.